msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenSilentSupplier.java

```diff
@@ -53,29 +55,9 @@ AuthenticationResult execute() throws Exception {
                 clientApplication.serviceBundle().getServerSideTelemetry().incrementSilentSuccessfulCount();
             }
 
-            //Determine if the current token needs to be refreshed according to the refresh_in value
-            long currTimeStampSec = new Date().getTime() / 1000;
-            boolean afterRefreshOn = res.refreshOn() != null && res.refreshOn() > 0 &&
-                    res.refreshOn() < currTimeStampSec && res.expiresOn() >= currTimeStampSec;
-
-            if (silentRequest.parameters().forceRefresh() || afterRefreshOn || StringHelper.isBlank(res.accessToken())) {
-
-                //As of version 3 of the telemetry schema, there is a field for collecting data about why a token was refreshed,
-                // so here we set the telemetry value based on the cause of the refresh
-                if (silentRequest.parameters().forceRefresh()) {
-                    clientApplication.serviceBundle().getServerSideTelemetry().getCurrentRequest().cacheInfo(
-                            CacheTelemetry.REFRESH_FORCE_REFRESH.telemetryValue);
-                } else if (afterRefreshOn) {
-                    clientApplication.serviceBundle().getServerSideTelemetry().getCurrentRequest().cacheInfo(
-                            CacheTelemetry.REFRESH_REFRESH_IN.telemetryValue);
-                } else if (res.expiresOn() < currTimeStampSec) {
-                    clientApplication.serviceBundle().getServerSideTelemetry().getCurrentRequest().cacheInfo(
-                            CacheTelemetry.REFRESH_ACCESS_TOKEN_EXPIRED.telemetryValue);
-                } else if (StringHelper.isBlank(res.accessToken())) {
-                    clientApplication.serviceBundle().getServerSideTelemetry().getCurrentRequest().cacheInfo(
-                            CacheTelemetry.REFRESH_NO_ACCESS_TOKEN.telemetryValue);
-                }
+            shouldRefresh = shouldRefresh(silentRequest.parameters(), res);
```

Comment:
The reason for the cache refresh should be captured in logging satements.

Reply:
Added a logging statement for each refresh reason (at the same logging level as the 'returned token from cache' message)